File: src/test/java/example/example/factory/PageinstancesFactory.java
package example.example.factory;

import java.lang.reflect.InvocationTargetException;

import com.microsoft.playwright.Page;

import example.example.context.WebDriverContext;
import example.example.pages.BasePage;

public class PageinstancesFactory {

	public static <T extends BasePage> T getInstance(Class<T> type) {
		try {
			return type.getConstructor(Page.class).newInstance(WebDriverContext.getDriver());
		} catch (InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException
				| NoSuchMethodException | SecurityException e) {
			e.printStackTrace();
			return null;
		}
	}
}

File: src/test/python/example/example/factory/page_instances_factory.py
from typing import Type, TypeVar

from example.example.context.web_driver_context import get_driver
from example.example.pages.base_page import BasePage

T = TypeVar("T", bound=BasePage)


def get_instance(page_class: Type[T]) -> T:
    driver = get_driver()
    return page_class(driver)
